cypress/integration/Scheduler/Scheduler_filter_spec.js

```diff
@@ -0,0 +1,20 @@
+describe("Tests Scheduler Filter if react-select throws 'options.reduce is not a function' error", () => {
+    before(() => {
+        cy.visit("scheduler")
+        cy.login("danielhuang@blah.com", "password")
+    })
+    it("Should go to scheduler page", () => {
+        cy.visit("scheduler")
+        cy.get("[data-cy=scheduler-header-text]").contains("Scheduler")
+    })
+    it("Checks if you can enter your own filter", () => {
+        cy.get("[data-cy=schduler-select-filter-box]").click()
+        cy.get("#filter-instructor-select").type("TESTINGSTESTING").type('{enter}')
+        cy.get("#filter-instructor-select").click(0, 20)
+        cy.get("#filter-course-select").type("TEST2TEST2TEST2").type('{enter')
+        cy.get("#filter-instructor-select").click(0, 20)
+        cy.get("#filter-instructor-select").should('have.value', '')
+        cy.get("#filter-instructor-select").click(0, 20)
+        cy.get("#filter-course-select").should('have.value', '')
```

Comment:
Try to use the `data-cy` attribute when selecting for specific elements

Reply:
Valid, but we are using react-select for these elements. I've tried: 

```
<ReactSelect
      data-cy="filter-instructor-select"
      className={"session-filter-select"}
      placeholder={"Filter Instructor..."}
      value={InstructorValue}
      options={InstructorOptions}
      onChange={onInstructorSelect}
      clearable
      isMulti
     />
```

This doesn't work so I've added the id prop and that works. Let me know if you have a solution to test React-admin.

Reply:
So what you should be able to do is traverse the Virtual DOM if you add a data-cy attribute to the parent. You would use  something like .children() to traverse it. Source [https://docs.cypress.io/api/commands/children.html#Syntax](https://docs.cypress.io/api/commands/children.html#Syntax). There is also .next() [https://docs.cypress.io/api/commands/next.html#Syntax](https://docs.cypress.io/api/commands/next.html#Syntax)

Reply:
Also should lines 12-18 have more assertions so we could actually test things? I.E
```
cy.get("#filter-instructor-select").type("TESTINGSTESTING").type('{enter}').should("have.value", '{enter}')
```

also, fix line 14.
```
 cy.get("#filter-course-select").type("TEST2TEST2TEST2").type('{enter} <---')
```